### ISSUE TITLE
False positive unused export reported in monorepo

### ISSUE THREAD

**Issue by kgregory** [webpro-nl/knip#719]
I have a monorepo that is resulting in a false positive for unused exports with the latest version of `knip` (version `5.25.1`)

I have pared our repository down to a greatly simplified [reproduction repo](https://github.com/kgregory/knip-repro-unused-export) consisting of the following:

- `@example/tsconfig`: A package to share a common TypeScript config
- `@example/app_common.ui`: A package that exports two react components (SVG icons from [@mui/icons-material](https://mui.com/material-ui/material-icons/))
- `@example/app.repro`: A package for a react app that imports and renders both of these components (that's all it does)

The issue:

- `@example/app.repro` imports `BarcodeScannerIcon` and `CachedIcon` from `@example/app_common.ui` and renders them both.
- knip returns a false positive, indicating that `BarcodeScannerIcon` is reported as unused.

```
Unused exports (1)
BarcodeScannerIcon  unknown  apps/app_common/ui/src/icons/BarcodeScannerIcon.tsx:5:14
```

Possibly worth noting:

- `BarcodeScannerIcon`, which is reported as unused, uses MUI's `SvgIcon` to build the custom SVG icon
- `CachedIcon`, which is not reported, is directly exported from `@mui/icons-material`

**Comment by webpro** [comment]
Thanks for the clear repro. You've hit quite an edge case, but ultimately the culprit is that there's no root `tsconfig.json` and `compilerOptions.moduleResolution: "bundler"` wasn't available/set where it should.

Currently you can do [--isolate-workspaces](https://knip.dev/reference/cli#--isolate-workspaces) and the next release of Knip will just always set that `moduleResolution` to "bundler" (can't yet think of situations where that could lead to issues).

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.25.2. See [Release 5.25.2](https://github.com/webpro-nl/knip/releases/tag/5.25.2) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._

**Comment by kgregory** [comment]
Thanks for turning this around so quickly!
